### ISSUE TITLE
`changefeeds.change_master` and `polyglot/changefeeds/edge` test failures

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4586]
Both tests are regularly failing for me on `next`:
```
FAIL changefeeds.change_master
  File "/home/ssd3/daniel/rethinkdb/test/changefeeds/change_master.test", line 90, in test_change_master
    self.assertRaises(self.r.RqlRuntimeError, pointFeed.next, wait=.2)
AssertionError: RqlRuntimeError not raised
```

```
<<< end Failed: polyglot/changefeeds/edge.js

== Failed 1 test (of 124) in 161.81 seconds!
	polyglot/changefeeds/edge.js
```

@mlucy could you take a look at these please?
(also mentioning @larkost )

**Comment by danielmewes** [comment]
Full `test-runner` output for the edge test:
```
$ ./test-runner -i js polyglot/changefeeds/edge
Using rethinkdb binary /home/ssd3/daniel/rethinkdb/build/release_clang/rethinkdb
	JavaScript interpreter: /usr/local/bin/node, driver: /home/ssd3/daniel/rethinkdb/drivers/javascript
== Starting: polyglot/changefeeds/edge.js (T+ 0.3 sec)
>>> Failed polyglot/changefeeds/edge.js with exit code 1 after 2.7 sec + 0.1 sec setup (T+ 3.3 sec)

Using RethinkDB client from: /home/ssd3/daniel/rethinkdb/build/packages/js
TEST FAILURE:  polyglot/changefeeds/edge.js/38
     SOURCE:   postmap_changes1
     EXPECTED: err('RqlNonExistenceError', "No attribute `dummy` in object.")
     RESULT:   RqlRuntimeError: No attribute `dummy` in object:
{
	"new_val":	{
		"id":	"627de7d4-68b6-4366-9",
		"value":	"2"
	},
	"old_val":	null
}
Stack:
RqlRuntimeError: No attribute `dummy` in object:
{
	"new_val":	{
		"id":	"627de7d4-68b6-4366-9",
		"value":	"2"
	},
	"old_val":	null
} in:
r.db("test").table("polyglot_changefeeds_edge_js").changes({squash: false}).map(r.branch(r.row("new_val")("value").lt("2"), r.row, r.row("dummy"))).limit(13)
                                                                                                                                   ^^^^^^^^^^^^^^            
    at RqlRuntimeError.RqlServerError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:39:13)
    at new RqlRuntimeError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:51:50)
    at mkErr (/home/ssd3/daniel/rethinkdb/build/packages/js/util.js:173:10)
    at Feed.IterableResult._promptNext (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:165:16)
    at Feed.IterableResult._addResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:85:12)
    at TcpConnection.Connection._processResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:132:16)
    at TcpConnection.Connection._data (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:116:12)
    at Socket.<anonymous> (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:537:32)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:736:14)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:408:10)
    at emitReadable (_stream_readable.js:404:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
    at TCP.onread (net.js:511:21)

TEST FAILURE:  polyglot/changefeeds/edge.js/39
     SOURCE:   postmap_changes2
     EXPECTED: err('RqlNonExistenceError', "Index out of bounds:" + " 1")
     RESULT:   RqlRuntimeError: Index out of bounds: 1
Stack:
RqlRuntimeError: Index out of bounds: 1 in:
r.db("test").table("polyglot_changefeeds_edge_js").changes({squash: false}).map(r.branch(r.row("new_val")("value").lt("2"), r.row, r([]).nth(1))).limit(13)
                                                                                                                                   ^^^^^^^^^^^^            
    at RqlRuntimeError.RqlServerError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:39:13)
    at new RqlRuntimeError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:51:50)
    at mkErr (/home/ssd3/daniel/rethinkdb/build/packages/js/util.js:173:10)
    at Feed.IterableResult._promptNext (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:165:16)
    at Feed.IterableResult._addResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:85:12)
    at TcpConnection.Connection._processResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:132:16)
    at TcpConnection.Connection._data (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:116:12)
    at Socket.<anonymous> (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:537:32)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:736:14)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:408:10)
    at emitReadable (_stream_readable.js:404:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
    at TCP.onread (net.js:511:21)

TEST FAILURE:  polyglot/changefeeds/edge.js/40
     SOURCE:   postmap_changes3
     EXPECTED: err('RqlUserError', "dummy")
     RESULT:   RqlRuntimeError: dummy
Stack:
RqlRuntimeError: dummy in:
r.db("test").table("polyglot_changefeeds_edge_js").changes({squash: false}).map(function(var_10) { return r.branch(r.random().gt(0.5), var_10, r.error("dummy")); }).limit(100)
                                                                                                                                               ^^^^^^^^^^^^^^^^                
    at RqlRuntimeError.RqlServerError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:39:13)
    at new RqlRuntimeError (/home/ssd3/daniel/rethinkdb/build/packages/js/errors.js:51:50)
    at mkErr (/home/ssd3/daniel/rethinkdb/build/packages/js/util.js:173:10)
    at Feed.IterableResult._promptNext (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:165:16)
    at Feed.IterableResult._addResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:85:12)
    at TcpConnection.Connection._processResponse (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:132:16)
    at TcpConnection.Connection._data (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:116:12)
    at Socket.<anonymous> (/home/ssd3/daniel/rethinkdb/build/packages/js/net.js:537:32)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:736:14)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:408:10)
    at emitReadable (_stream_readable.js:404:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
    at TCP.onread (net.js:511:21)

Failed 3 tests
Cleaning up

<<< end Failed: polyglot/changefeeds/edge.js

== Failed 1 test (of 1) in 3.43 seconds!
	polyglot/changefeeds/edge.js
```

**Comment by danielmewes** [comment]
...actually that one looks like a problem with the error types.

**Comment by danielmewes** [comment]
@larkost has been looking into the `edge` test. Reassigning. Feel free to assign this back if there turn out to be open questions that are not on the tests' side of things.

**Comment by larkost** [comment]
The `polyglot/changefeeds/edge.js` part of this is solved in #4590. Looking at `changefeeds.change_master` now.

**Comment by larkost** [comment]
I don't see the `changefeeds.change_master` after trying a bit, so we are assuming that this issue is gone.
